crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -0,0 +1,188 @@
+use anyhow::{
+    anyhow,
+    bail,
+    Context,
+};
+use penumbra_storage::Storage;
+use tendermint::abci::{
+    request,
+    response,
+    ConsensusRequest,
+    ConsensusResponse,
+};
+use tokio::sync::mpsc;
+use tower_abci::BoxError;
+use tower_actor::Message;
+use tracing::{
+    instrument,
+    Instrument,
+};
+
+use crate::{
+    app::App,
+    genesis::GenesisState,
+};
+
+pub(crate) struct Consensus {
+    queue: mpsc::Receiver<Message<ConsensusRequest, ConsensusResponse, tower::BoxError>>,
+    storage: Storage,
+    app: App,
+}
+
+impl Consensus {
+    pub(crate) fn new(
+        storage: Storage,
+        app: App,
+        queue: mpsc::Receiver<Message<ConsensusRequest, ConsensusResponse, tower::BoxError>>,
+    ) -> Self {
+        Self {
+            queue,
+            storage,
+            app,
+        }
+    }
+
+    pub(crate) async fn run(mut self) -> Result<(), tower::BoxError> {
+        while let Some(Message {
+            req,
+            rsp_sender,
+            span,
+        }) = self.queue.recv().await
+        {
+            // The send only fails if the receiver was dropped, which happens
+            // if the caller didn't propagate the message back to tendermint
+            // for some reason -- but that's not our problem.
+            rsp_sender.send(self.handle_request(req).instrument(span).await);
```

Comment:
I broke-fixed a clippy lint. Fix:

Split this into 2 steps:

1. Match on `self.handle_request`.
    a. if an error, emit a `warn!(parent: &span, ...)` event with the error
2. `if let Err(e) = rsp_sender.send` with the ok-value, and then `warn!` in the same span

Reply:
done